module/platform/cm740/HardwareIO/src/HardwareIO.hpp

```diff
@@ -37,69 +37,77 @@ namespace module::platform::cm740 {
-
+            /// @brief Battery voltage information.
             struct Battery {
                 Battery() = default;
-                float chargedVoltage{0.0f};
-                float nominalVoltage{0.0f};
-                float flatVoltage{0.0f};
+                float charged_voltage{0.0f};
+                float nominal_voltage{0.0f};
+                float flat_voltage{0.0f};
             } battery;
-        } config;
+        } cfg;
 
         struct ServoState {
-            // True if we need to write new values to the hardware
+            /// @brief True if new values should be written to the hardware.
             bool dirty = false;
 
-            // True if we simulate where we think the servos should be
-            // Note that we still write the commands to hardware
+            /// @brief True if servo positions are simulated.
+            /// @details Note that commands can still be written to the hardware.
             bool simulated = false;
 
-            bool torqueEnabled = true;
-
-            // Cached values that are never read
-            float pGain        = 32.0 / 255.0;
-            float iGain        = 0;
-            float dGain        = 0;
-            float movingSpeed  = 0;
-            float goalPosition = 0;
-            float torque       = 0;  // 0.0 to 1.0
-
-            // Values that are either simulated or read
-            float presentPosition = 0;
-            float presentSpeed    = 0;
-            float load            = 0;
-            float voltage         = 10;
-            float temperature     = 40;
+            /// @brief Whether the servo torque is enabled.
+            bool torque_enabled = true;
+
+            /// @brief Proportional gain. Cached value that is never read.
+            float p_gain = 32.0 / 255.0;
+            /// @brief Integral gain. Cached value that is never read.
+            float i_gain = 0;
+            /// @brief Derivative gain. Cached value that is never read.
+            float d_gain = 0;
+            /// @brief Speed of the servo. Cached value that is never read.
+            float moving_speed = 0;
+            /// @brief Position the servo is attempting to reach. Cached value that is never read.
+            float goal_position = 0;
+            /// @brief Torque of the servo. Cached value that is never read.
+            /// @details Ranges from 0.0 to 1.0.
+            float torque = 0;
+
+            /// @brief Current angle position of the servo. Simulated or read.
+            float present_position = 0;
+            /// @brief Current speed of the servo. Simulated or read.
+            float present_speed = 0;
+            /// @brief Current load of the servo. Simulated or read.
+            float load = 0;
+            /// @brief Current voltage of the servo. Simulated or read.
+            float voltage = 10;
+            /// @brief Current temperature of the servo. Simulated or read.
+            float temperature = 40;
```

Comment:
technically you are initialising `float`s with `int`s here .... although the scope creep in this PR is already pretty insane